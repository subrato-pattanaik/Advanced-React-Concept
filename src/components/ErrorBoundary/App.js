import * as React from 'react'
import {ErrorBoundary as ReactErrorBoundary} from 'react-error-boundary'

function ErrorFallback({error}) {
  return (
    <div role="alert">
      <p>Something went wrong:</p>
      <pre style={{color: 'red'}}>{error.message}</pre>
    </div>
  )
}
function Greeting({subject}) {
  return <div>Hello {subject.toUpperCase()}</div>
}
function Farewell({subject}) {
  return <div>Goodbye {subject.toUpperCase()}</div>
}
function App() {
  return (
    <ReactErrorBoundary FallbackComponent={ErrorFallback}>
      <Greeting />
      <Farewell />
    </ReactErrorBoundary>
  )
}

export default App
