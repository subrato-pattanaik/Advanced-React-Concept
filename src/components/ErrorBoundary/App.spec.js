import {render, screen} from '@testing-library/react'
import App from './App'

describe('Testing App Component', () => {
  //we had mock the console.error message as when the error's is thrown
  //in the program a bunch of console.error are called even though Error Boundary
  //handles it.
  beforeEach(() => {
    jest.spyOn(console, 'error').mockImplementation(() => {})
    // console.error.mockImplementation(() => {})
  })

  afterEach(() => {
    console.error.mockRestore()
  })

  it('should render alert message', () => {
    //Arrange
    render(<App />)
    //Assertion
    expect(screen.getByRole('alert')).toBeInTheDocument()
    expect(console.error).toHaveBeenCalled()
    expect(console.error).toHaveBeenCalledTimes(4)
  })

  it('should render correctly', () => {
    const {container} = render(<App />)
    expect(container).toMatchSnapshot()
  })
})
