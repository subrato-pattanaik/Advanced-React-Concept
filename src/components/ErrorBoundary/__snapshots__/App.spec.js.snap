// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App Component should render correctly 1`] = `
<div>
  <div
    role="alert"
  >
    <p>
      Something went wrong:
    </p>
    <pre
      style="color: red;"
    >
      TypeError
       : 
      Cannot read property 'toUpperCase' of undefined
    </pre>
  </div>
</div>
`;
